### PR TITLE
feat(core): Implement parsing for register transaction

### LINKED ISSUE #421: Implement parsing for register transaction

### PR DESCRIPTION
**Pull request by dauTT**
Closes #421 

Note: please be aware that this PR has a dependency with PR #420 and can't be merged until PR #420 is fixed.

### CODE REVIEW

File: packages/neon-core/src/tx/transaction/RegisterTransaction.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  public name: string;
+  public amount: Fixed8;
+  public precision: number;
+  public owner: string;
+  public admin: string;
+
+  public readonly type: TransactionType = TransactionType.RegisterTransaction;
+
+  public constructor(obj: Partial<RegisterTransactionLike> = {}) {
+    super(Object.assign({ version: TX_VERSION.REGISTER }, obj));
```
Comment (snowypowers) [inline packages/neon-core/src/tx/transaction/RegisterTransaction.ts]: forgot to add TX_VERSION.REGISTER into consts.

Comment (dauTT) [reply packages/neon-core/src/tx/transaction/RegisterTransaction.ts]: Ok, I will add it.

**Comment by hal0x2328** [comment]
300 points dauTT
100 points snowypowers for review

**Comment by lllwvlvwlll** [comment]
https://neoscan.io/transaction/589328c622e101065f6717d61a332f93aac71cc68b665e194ef8c874a7e26669

**Pull request merged**